File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.swift
import Foundation

func solution(_ arr:[Int], _ queries:[[Int]]) -> [Int] {
    var result: [Int] = []
    queries.forEach { innerArr in
        let num = arr[innerArr[0]...innerArr[1]].sorted().first { $0 > innerArr[2] } ?? -1
        result.append(num)
    }
    
    return result
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;

    for (auto query : queries) {
        int s = query[0], e = query[1], k = query[2], val = 1000005;

        for (int i = s; i <= e; i++) {
            if (arr[i] > k) {
                val = min(val, arr[i]);
            }
        }
        (val != 1000005) ? answer.push_back(val) : answer.push_back(-1);
    }

    return answer;
}